### PR TITLE
Implement the `chain_id` webb proposal for Cosmos SDK blockchains

### LINKED ISSUE #39: [TASK] Compute numerical chain_id (u32) for cosmos chains

### PR DESCRIPTION
**Pull request by duguorong009**
Summary of changes
Changes introduced in this pull request
   - Implement the new `chain_id` construction proposal from #39 
   - Update the corresponding contracts
       `anchor`, `vanchor`, `signature-bridge` & `anchor-handler` contracts

Reference issue to close (if applicable)
Closes #39

### CODE REVIEW

File: contracts/anchor/src/state.rs [hunk 1]
```diff
@@ -30,10 +30,9 @@ pub const NULLIFIERS: Map<Vec<u8>, bool> = Map::new("used_nullifers");
 //       `chain_id`(blockchain-unique ID) obtained inside the contract.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
 pub struct Anchor {
-    pub chain_id: u64,           // ChainID of underlying blockchain(Temporary field)
-    pub handler: Addr,           // Address of "anchor-handler", which add/updte the `edge` info
-    pub proposal_nonce: u32,     // Proposal nonce
-    pub deposit_size: Uint128,   // Minimum `deposit` amount for tx
+    pub handler: Addr, // Address of "anchor-handler", which add/updte the `edge` info
```
Comment (drewstone) [inline contracts/anchor/src/state.rs:33]: Can we put comments above the code? On the side is bad style.

**Review by drewstone — CHANGES_REQUESTED** [review]

File: contracts/anchor/src/state.rs (hunk 1 above)
Comment (drewstone) [reply contracts/anchor/src/state.rs:33]: This should happen for all of these parameters and please use three slashes `///`.

File: packages/protocol_cosmwasm/src/utils.rs [hunk 2]
```diff
@@ -28,6 +29,23 @@ pub fn compute_chain_id_type(chain_id: u64, chain_type: &[u8]) -> u64 {
     u64::from_be_bytes(buf)
 }
 
+// Computes the numeric "chain_id" from string one.
```
Comment (drewstone) [inline packages/protocol_cosmwasm/src/utils.rs:32]: Three slashes here `///`

**Review by drewstone — CHANGES_REQUESTED** [review]

File: packages/protocol_cosmwasm/src/utils.rs (hunk 2 above)
Comment (duguorong009) [reply packages/protocol_cosmwasm/src/utils.rs]: Done https://github.com/webb-tools/protocol-cosmwasm/pull/57/commits/ac2759ac676c3702d19f579ca740bb36e18481ef

File: contracts/anchor/src/state.rs (hunk 1 above)
Comment (duguorong009) [reply contracts/anchor/src/state.rs]: Done https://github.com/webb-tools/protocol-cosmwasm/pull/57/commits/ac2759ac676c3702d19f579ca740bb36e18481ef

**Review by shekohex — APPROVED** [review]
LGTM!
Just one more thing is to remove these wasm files from the code.

**Comment by drewstone** [comment]
I think it's OK to have the wasm files for now.

**Pull request merged**